templates/etc/network/interfaces.d/iface.j2

```diff
@@ -22,6 +23,20 @@
 {%   endif %}
 {%   set ifupdown__tpl_ipv4_addresses = ifupdown__tpl_addresses | ipv4('host/prefix') %}
 {%   set ifupdown__tpl_ipv6_addresses = ifupdown__tpl_addresses | ipv6('host/prefix') %}
+{%   if interface.gateway|d() and interface.gateway %}
+{%     if interface.gateway is string %}
+{%       set ifupdown__tpl_gateways = [ interface.gateway ] %}
+{%     else %}
+{%       set ifupdown__tpl_gateways = interface.gateway %}
+{%     endif %}
+{%   endif %}
+{%   if interface.gateways|d() and interface.gateways %}
+{%     if interface.gateways is string %}
+{%       set ifupdown__tpl_gateways = (ifupdown__tpl_gateways + [ interface.gateways ]) %}
+{%     else %}
+{%       set ifupdown__tpl_gateways = (ifupdown__tpl_gateways + interface.gateways) %}
```

Comment:
That would need to be iterated, since you don't want to add a list as an item, but just the elements.